### PR TITLE
fix up spacing in `f a#:b` (instead of `f a #: b`)

### PR DESCRIPTION
**Pull request by v-gb**
It is very confusing to print `f a #: b` as ocamlformat does before this PR, as `#:` has higher priority than function application. So I change this to print `f a#:b`, same as method calls.

I'll admit the existing tests don't look improved, but I claim that's because they don't represent normal uses of operators starting with `#`.

### CODE REVIEW

**Review by Julow — APPROVED** [review]
Awesome ! Can you add a changelog entry ?

The output of test-branch looks good (except for the unrelated bug https://github.com/ocaml-ppx/ocamlformat/issues/2582).

**Comment by v-gb** [comment]
Thanks!

> Awesome ! Can you add a changelog entry ?

Done.

File: test/passing/tests/infix_bind-break.ml.ref [hunk 1]
```diff
@@ -232,10 +232,10 @@ let encoder f =
   stagged @@ fun x k : t -> field_encode (f.fget x) k
 
 let default =
-  command ## hasPermission #= (fun ctx -> foooooooooooooooooo fooooooooooo) ;
-  command ## hasPermission
-  #= (fun ctx ->
-  foooooooooooooooooo fooooooooooo foooooo fooooooooo foooooooooo ) ;
+  command##hasPermission#=(fun ctx -> foooooooooooooooooo fooooooooooo) ;
```
[the remark this answers is not in the document]
Comment (v-gb) [reply test/passing/tests/infix_bind-break.ml.ref:235]: Right, I don't plan on changing anything here. I'm not sure this code sample even comes from a real codebase.

**Comment by Julow** [comment]
Thanks!

**Comment by giltho** [comment]
> I'll admit the existing tests don't look improved, but I claim that's because they don't represent normal uses of operators starting with #.

@v-gb The code in the tests is in fact my code 😭 
https://github.com/GillianPlatform/Gillian/blob/4e4665455697f5d521fd28409a02d61f9bf74ad6/GillianCore/GIL_Syntax/Formula.ml#L370-L445

I'm not quite sure what "normal use of operators starting with #", given that there exist only 5 of those in the wild [looking at sherlodoc](https://doc.sherlocode.com/?q=%23)

**Comment by giltho** [comment]
> Is there a reason you were using specifically operators starting with '#'?

`#` is kind of convenient for us as it is the same character used in the concrete syntax to denote symbolic things. We might end up changing, but none of the other possible characters would have this feature, and note that `&&#` is not a valid name for an infix operator.

But as I said in #2638, I do understand why this was introduced, but possibly these kinds of changes should be introduced with some control over them.

**Comment by v-gb** [comment]
If that helps, I've updated the whole repo to other operators (which can't be done with a simple sed invocation, because the operators after have different precedence. This was done with an AST to AST rewriting, so the precedence is trivially preserved), [changing #op to %op](https://github.com/GillianPlatform/Gillian/compare/master...v-gb:Gillian:push-ltkxppvqryuv), or [changes #op to op@](https://github.com/GillianPlatform/Gillian/compare/master...v-gb:Gillian:push-uuorntzwvoqv).

A couple of things should be manually cleaned up (comments being moved to wrong places, as ocamlformat is wont to do), but that's much easier than manually updating the rest of the code while being sure not to modify the parenthesization.

**Pull request merged**